runtime/doc/lua.txt

```diff
@@ -2937,6 +2930,34 @@ vim.fs.find({names}, {opts})                                   *vim.fs.find()*
         (`string[]`) Normalized paths |vim.fs.normalize()| of all matching
         items
 
+vim.fs.findroot({bufnr}, {marker})                         *vim.fs.findroot()*
+    Find the first parent directory containing a specific "marker", relative
+    to a buffer's directory.
+
+    Example: >lua
+        -- Find the root of a Python project
+        vim.fs.findroot(0, {'pyproject.toml', 'setup.py' })
+
+        -- Find the root of a git repository
+        vim.fs.findroot(0, '.git')
+
+        -- Find the parent directory containing any file with a .csproj extension
+        vim.fs.findroot(0, function(name, path)
+          return name:match('%.csproj$') ~= nil
+        end)
+<
+
+    Parameters: ~
+      • {bufnr}   (`integer`) Buffer number, or 0 for current buffer.
+      • {marker}  (`string|string[]|fun(name: string, path: string): boolean`)
+                  A marker, or list of markers, to search for. If a function,
+                  the function is called for each evaluated item and should
+                  return true if {name} and {path} are a match.
+
+    Return: ~
+        (`string?`) Directory containing one of the given markers, or nil if
```

Comment:
IMO `Directory` makes it sound as if this was returning an object.

```suggestion
        (`string?`) Directory name containing one of the given markers, or nil if
```

Reply:
Maybe "Name of directory containing ..." would make more sense?

Reply:
d917f67c30d77c2c88848c81d8e964aa57b5521b